src/algebra/order/field.lean

```diff
@@ -725,4 +726,34 @@ lemma one_div_pow_mono (a1 : 1 ≤ a) : monotone (λ n : ℕ, order_dual.to_dual
 lemma one_div_pow_strict_mono (a1 : 1 < a) : strict_mono (λ n : ℕ, order_dual.to_dual 1 / a ^ n) :=
 λ m n, one_div_pow_lt_one_div_pow_of_lt a1
 
+/-! ### Results about `has_lub` and `has_glb` -/
+
+lemma is_lub_mul {s : set α} (hc : 0 ≤ a) (hs : is_lub s b) :
+  is_lub {x | ∃ y ∈ s, a * y = x} (a * b) :=
+begin
+  rcases lt_or_eq_of_le hc with hc | rfl,
+  { rw ← (order_iso.mul_left₀ _ hc).is_lub_image' at hs,
+    convert hs using 1,
+    ext x,
+    simp },
+  { convert is_lub_singleton using 1,
+    ext x,
+    have : s.nonempty ∧ 0 = x ↔ x = 0 := by rw [and_iff_right hs.nonempty, eq_comm],
+    simpa using this },
+end
```

Comment:
This statement feels more useful, and the proof is much shorter:
```suggestion
lemma is_lub_mul {s : set α} (hc : 0 ≤ a) (hs : is_lub s b) :
  is_lub ((λ b, a * b) '' s) (a * b) :=
begin
  rcases lt_or_eq_of_le hc with hc | rfl,
  { exact (order_iso.mul_left₀ _ hc).is_lub_image'.2 hs, },
  { simp_rw zero_mul,
    rw hs.nonempty.image_const,
    exact is_lub_singleton },
end
```

Reply:
With #11021 you could replace the `simp_rw` with a `rw`, but it doesn't really matter.